# not sure the reason yet. Keep for a while and verify later.
#
# comments added by Kevin Tian <kevin.tian@intel.com>, 2010-07-21

Upstream-Status: Pending

diff -ruN module-init-tools-3.12.orig/modprobe.8 module-init-tools-3.12/modprobe.8
--- module-init-tools-3.12/modprobe.8.orig
+++ module-init-tools-3.12/modprobe.8
@@ -27,6 +27,7 @@
 (see \fBmodprobe.conf\fR(5)). \fBmodprobe\fR will also use module
 options specified on the kernel command line in the form of
 <module>\&.<option>\&.
+All files in the \fI/etc/modprobe.d/arch/\fR directory are ignored.
 .PP
 Note that unlike in 2.4 series Linux kernels (which are not supported
 by this tool) this version of \fBmodprobe\fR does not
--- module-init-tools-3.12/modprobe.c.orig
+++ module-init-tools-3.12/modprobe.c
@@ -1052,6 +1052,10 @@
 	DIR *dir;
 	int ret = 0;
 
+	/* ignore everything in this directory */
+	if (streq(filename, "/etc/modprobe.d/arch"))
+		return 1;
+
 	dir = opendir(filename);
 	if (dir) {
 		struct file_entry {
